include/NotificationsManager.php

```diff
@@ -28,109 +32,762 @@ public function __construct() {
-            $filter_str[] = sprintf("`%s` = '%s'", $column, dbesc($value));
-        }
-        if (count($filter_str)>0) {
-            $filter_sql = "AND ".implode(" AND ", $filter_str);
-        }
-        
-        $aOrder = explode(" ", $order);
-        $asOrder = array();
-        foreach($aOrder as $o) {
-            $dir = "asc";
-            if ($o[0]==="-") {
-                $dir = "desc";
-                $o = substr($o,1);
-            }
-            if ($o[0]==="+") {
-                $dir = "asc";
-                $o = substr($o,1);
-            }
-            $asOrder[] = "$o $dir";
-        }
-        $order_sql = implode(", ", $asOrder);
-        
-        if ($limit!="") $limit = " LIMIT ".$limit;
-        
-		$r = q("SELECT * FROM `notify` WHERE `uid` = %d $filter_sql ORDER BY $order_sql $limit",
+	private function _set_extra($notes) {
+		$rets = array();
+		foreach($notes as $n) {
+			$local_time = datetime_convert('UTC',date_default_timezone_get(),$n['date']);
+			$n['timestamp'] = strtotime($local_time);
+			$n['date_rel'] = relative_date($n['date']);
+				$n['msg_html'] = bbcode($n['msg'], false, false, false, false);
+				$n['msg_plain'] = explode("\n",trim(html2plain($n['msg_html'], 0)))[0];
+
+			$rets[] = $n;
+		}
+		return $rets;
+	}
+
+
+	/**
+	 * @brief Get all notifications for local_user()
+	 *
+	 * @param array $filter optional Array "column name"=>value: filter query by columns values
+	 * @param string $order optional Space separated list of column to sort by. prepend name with "+" to sort ASC, "-" to sort DESC. Default to "-date"
+	 * @param string $limit optional Query limits
+	 *
+	 * @return array of results or false on errors
+	 */
+	public function getAll($filter = array(), $order="-date", $limit="") {
+		$filter_str = array();
+		$filter_sql = "";
+		foreach($filter as $column => $value) {
+			$filter_str[] = sprintf("`%s` = '%s'", $column, dbesc($value));
+		}
+		if (count($filter_str)>0) {
+			$filter_sql = "AND ".implode(" AND ", $filter_str);
+		}
+
+		$aOrder = explode(" ", $order);
+		$asOrder = array();
+		foreach($aOrder as $o) {
+			$dir = "asc";
+			if ($o[0]==="-") {
+				$dir = "desc";
+				$o = substr($o,1);
+			}
+			if ($o[0]==="+") {
+				$dir = "asc";
+				$o = substr($o,1);
+			}
+			$asOrder[] = "$o $dir";
+		}
+		$order_sql = implode(", ", $asOrder);
+
+		if ($limit!="") $limit = " LIMIT ".$limit;
+
+			$r = q("SELECT * FROM `notify` WHERE `uid` = %d $filter_sql ORDER BY $order_sql $limit",
+				intval(local_user())
+			);
+		if ($r!==false && count($r)>0) return $this->_set_extra($r);
+		return false;
+	}
+
+	/**
+	 * @brief Get one note for local_user() by $id value
+	 *
+	 * @param int $id
+	 * @return array note values or null if not found
+	 */
+	public function getByID($id) {
+		$r = q("SELECT * FROM `notify` WHERE `id` = %d AND `uid` = %d LIMIT 1",
+			intval($id),
+			intval(local_user())
+		);
+		if($r!==false && count($r)>0) {
```

Comment:
I would suggest to use the function "dbm::is_result" - for all places in the code where you did similar changes.